static/app/components/sidebar/newOnboardingStatus.tsx

```diff
@@ -98,6 +99,17 @@ export function NewOnboardingStatus({
     });
   }, [isActive, totalRemainingTasks, organization]);
 
+  useEffect(() => {
+    if (pendingCompletionSeen && isActive && !hasMarkedUnseenTasksAsComplete.current) {
+      markDoneTaskAsComplete();
+      hasMarkedUnseenTasksAsComplete.current = true;
+    }
+
+    if (!pendingCompletionSeen || !isActive) {
+      hasMarkedUnseenTasksAsComplete.current = false;
+    }
+  }, [isActive, pendingCompletionSeen, markDoneTaskAsComplete]);
```

Comment:
Users can keep the sidebar open while completing tasks. Therefore, we shouldn't mark unseen tasks only when the sidebar is opened; we should also do so while it is active and users have completed tasks.